Index: remmina/src/remmina_file_editor.c
===================================================================
--- remmina.orig/src/remmina_file_editor.c
+++ remmina/src/remmina_file_editor.c
@@ -1945,7 +1945,8 @@ static void remmina_file_editor_init(Rem
 
 	gtk_dialog_set_default_response(GTK_DIALOG(gfe), GTK_RESPONSE_OK);
 	gtk_window_set_default_size(GTK_WINDOW(gfe), 800, 600);
-
+	gtk_window_set_position(GTK_WINDOW(gfe), GTK_WIN_POS_CENTER);
+	
 	g_signal_connect(G_OBJECT(gfe), "destroy", G_CALLBACK(remmina_file_editor_destroy), NULL);
 	g_signal_connect(G_OBJECT(gfe), "realize", G_CALLBACK(remmina_file_editor_on_realize), NULL);
 
